src/v/archival/ntp_archiver_service.cc

```diff
@@ -135,6 +135,8 @@ ss::future<> ntp_archiver::upload_loop() {
             if (backoff > max_backoff) {
                 backoff = max_backoff;
             }
+        } else {
```

Comment:
can we merge this with commit that introduces this part ?

Reply:
Sure, this code was actually moved from `service.cc` and then the bug was fixed. But it can be a single commit.